draft-ietf-rtcweb-jsep.xml

```diff
@@ -1977,13 +1977,205 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
+        The SDP is read,
+        line-by-line, and converted to a data structure that contains the
+        deserialized information. However, SDP supports various fields that are not relevant to JSEP applications, and so these fields can be ignored.
+        For each line, the implementation will first ensure it is syntactically
+        correct, and then either parse the data to store for later use,
+        or discard the line and move on. If the line is not well-formed, or
+        cannot be parsed as described, the specified the parser MUST stop with an error and reject the session decription. This ensures that implementations do not accidentally misinterpret ambiguous SDP.</t>
+
+        <section title="Session-Level Parsing">
+          <t>First, the session-level lines are checked and parsed. These
+            lines MUST occur in a specific order, and with a specific syntax, as defined in <xref target="RFC4566"/>, Section 5. </t>
+          <t>For non-attribute (a=) lines, their sequencing and syntax are checked,
+            as mentioned above, but all values MUST be discarded. For example,
+            because the use of ICE is mandatory, the c= line is superfluous and its value MUST be discarded; similarly, because session-level bandwidth is ambiguous when multiple media streams are present, a b= line at session
+            level is not useful and its value MUST be discarded.</t>
+          <t>Specific processing is applied for the following session-level
+            attribute (a=) lines:
+            <list style="symbols">
+              <t>Any a=group lines are parsed as specified in [TBD], and, if
+                valid, the group's semantics and mids are stored. [OPEN ISSUE: is
+                validation, e.g. of valid mids, done at this stage?]</t>
+              <t>If present, an a=ice-ufrag line is parsed as specified in
+                [TBD], and, if valid, the ufrag value is stored.</t>
+              <t>If present, an a=ice-pwd line is parsed as specified in
+                [TBD], and, if valid, the pwd value is stored.</t>
+              <t>If present, an a=ice-lite line is parsed as specified in
+                [TBD], and, if valid, a value indicating the presence of ice-lite
+                is stored.</t>
+              <t>If present, an a=ice-options is parsed as specified in
+                [TBD], and, if valid, the set of specified options is stored.</t>
+              <t>If present, any a=fingerprint lines are parsed as specified in
+                [TBD], and, if valid, the set of fingerprint and algorithm
+                values is stored.</t>
+              <t>If present, any a=setup attribute is parsed as specified in
+                [TBD], and, if valid, the value is stored.</t>
+              <t>TODO: msid-semantic and related values</t>
+            </list>
+            Once all the session-level lines have been parsed, processing
```

Comment:
I think we need to say what gets stuffed into the data structure which is then used in the 'apply' sections below. The createoffer/answer stuff is pretty specific, so I think we're already heading down this path a bit. But I held off on rewriting 4566 inside this spec.